src/main/java/uk/gov/companieshouse/company/profile/service/CompanyProfileService.java

```diff
@@ -102,17 +107,22 @@ public void updateInsolvencyLink(String contextId, String companyNumber,
                         contextId, "company_delta");
                 cpDocument.setUpdated(updated);
             }
-            companyProfileRepository.save(cpDocument);
-        } catch (IllegalArgumentException illegalArgumentEx) {
-            throw new BadRequestException(illegalArgumentEx.getMessage());
+            ApiResponse<Void> response = insolvencyApiService.invokeChsKafkaApi(
+                    contextId, companyNumber);
+
+            if (response.getStatusCode() == HttpStatus.OK.value()) {
+                companyProfileRepository.save(cpDocument);
+            } else {
+                String errorMessage = String.format(
+                        "Error occurred while calling /resource-changed with "
+                        + "contextId %s and company number %s", contextId, companyNumber);
+                logger.error(errorMessage);
+                throw new RuntimeException(errorMessage);
```

Comment:
All of this and more already happens in `src/main/java/uk/gov/companieshouse/company/profile/api/InsolvencyApiService.java` in the `invokeChsKafkaApi` method, no need to log and throw the same exception again.